SwiftUI View: TextField (#1350)
Codecademy/docs: content/swiftui/concepts/views/terms/textfield/textfield.md

---
Title: 'TextField'

> Description: 'It a view that allows user to insert their informations, type memos and edit the content.'

Reviewer: ```suggestion
Description: 'A view that allows the user to insert information, type memos, and edit the content.'
```

Subjects:
  - 'Mobile Development'
  - 'Computer Science'
Tags:
  - 'SwiftUI'
  - 'SwiftUI Views'
  - 'Views'
CatalogContent:
  - 'learn-swift'
  - 'paths/build-ios-apps-with-swiftui'
---

> The **`TextField`** is a view that allows user to fill in the field with the relative data. 

Reviewer: ```suggestion
The **`TextField`** is a view that allows the user to fill in the field with data. 
```

> The first parameter refers to the text that will be displayed in a lighter color inside the field. 

Reviewer: Lines 17 and 18 should be in the syntax section, and should refer back to placeholders in the ```pseudo code block.

The second parameter refers to what the input will be, it's a binding parameter so that swift will always know if and when this parameter changes.

## Syntax

```pseudo
var body: some View {
    TextField(_ title: StringProtocol, text: Binding<String>)
}
```

## Example

> In the example below, a `VStack` holds a series of `TextFields` and one `SecureField`. This example reassemble a sort of 'Sign Up' page.

Reviewer: ```suggestion
In the example below, a `VStack` holds a series of `TextFields` and one `SecureField`. This example resembles a 'Sign Up' page.
```

```swift
@State private var name = ""
@State private var lastName = ""
@State private var username = ""
@State private var email = ""
@State private var password = ""

var body: some View {
    VStack {
        TextField("Name", text: $name)
        
        TextField("Last Name", text: $lastName)
            .keyboardType(.default)
            
        TextField("Username", text: $username)
            .keyboardType(.asciiCapable)
            
        TextField("Email", text: $email)
            .keyboardType(.emailAddress)
            
        SecureField("Password", text: $password)
    }
    .padding()
}
```

> - The top 5 variables are market as private because it is important that those information do not leave the current page.

Reviewer: ```suggestion
- The top 5 variables are marked as private because it is important that the information does not leave the current page.
```

> - `TextField` not also has different keyboard types but also keyboard styles. If no type or style is specified Swift will display them as `(.default)`.

Reviewer: ```suggestion
- `TextField` not only has different keyboard types but also keyboard styles. If no type or style is specified Swift will display them as `(.default)`.
```

- `SecureField` is simply a `TextField` that instantly hides the character and is commonly used to setup passwords.

> - There are two images for this example, the first one represents an empty textfield and the second one a filled one.

Reviewer: ```suggestion
- There are two images for this example, the first one represents empty textfields and the second filled ones.
```

---
Title: 'TextField'
Description: 'A view that allows the user to insert information, type memos, and edit the content.'
Subjects:
  - 'Mobile Development'
  - 'Computer Science'
Tags:
  - 'SwiftUI'
  - 'SwiftUI Views'
  - 'Views'
CatalogContent:
  - 'learn-swift'
  - 'paths/build-ios-apps-with-swiftui'
---

The **`TextField`** is a view that allows the user to fill in the field with data. 

## Syntax

The first parameter refers to the text that will be displayed in a lighter color inside the field. 

> The second parameter refers to what the input will be, it's a binding parameter so that swift will always know if and when this parameter changes.

Reviewer 2: Can we move these to underneath the code block? Also if we can explicitly refer to the parameter names that would be great. Feel free to keep in paragraph or change to a bullet point format.

```pseudo
var body: some View {
    TextField(_ title: StringProtocol, text: Binding<String>)
}
```

## Example

In the example below, a `VStack` holds a series of `TextFields` and one `SecureField`. This example resembles a 'Sign Up' page.

```swift
@State private var name = ""
@State private var lastName = ""
@State private var username = ""
@State private var email = ""
@State private var password = ""

var body: some View {
    VStack {
        TextField("Name", text: $name)
        
        TextField("Last Name", text: $lastName)
            .keyboardType(.default)
            
        TextField("Username", text: $username)
            .keyboardType(.asciiCapable)
            
        TextField("Email", text: $email)
            .keyboardType(.emailAddress)
            
        SecureField("Password", text: $password)
    }
    .padding()
}
```

> - The top 5 variables are marked as private because it is important that the information does not leave the current page.
> - `TextField` not only has different keyboard types but also keyboard styles. If no type or style is specified Swift will display them as `(.default)`.
> - `SecureField` is simply a `TextField` that instantly hides the character and is commonly used to setup passwords.
> - There are two images for this example, the first one represents empty textfields and the second filled ones.

Reviewer 2: ```suggestion

- The top 5 variables are marked as private because it is important that the information does not leave the current page.
- `TextField` not only has different keyboard types but also keyboard styles. If no type or style is specified, Swift will display them as `(.default)`.
- `SecureField` is a `TextField` that instantly hides the character and is commonly used to setup passwords.

There are two images for this example. The first represents empty textfields and the second represents filled textfields:
```

This will display the following:

> ![empty TextField](https://raw.githubusercontent.com/Codecademy/docs/main/media/swiftui-textfield.png)
> ![filled TextField](https://raw.githubusercontent.com/Codecademy/docs/main/media/swiftui-textfield-fill.png)

Reviewer 2: ```suggestion
![Empty TextField](https://raw.githubusercontent.com/Codecademy/docs/main/media/swiftui-textfield.png)
![Filled TextField](https://raw.githubusercontent.com/Codecademy/docs/main/media/swiftui-textfield-fill.png)
```

> ![empty TextField](https://raw.githubusercontent.com/Codecademy/docs/main/media/swiftui-textfield.png)
> ![filled TextField](https://raw.githubusercontent.com/Codecademy/docs/main/media/swiftui-textfield-fill.png)

Reviewer 2: ```suggestion
![Empty TextField](https://raw.githubusercontent.com/Codecademy/docs/main/media/swiftui-textfield.png)
![Filled TextField](https://raw.githubusercontent.com/Codecademy/docs/main/media/swiftui-textfield-fill.png)
```

    TextField(_ title: StringProtocol, text: Binding<String>)
}
```

- `_ title:` refers to the text that will be displayed in a lighter color inside the field. 
- `text:` refers to what the input will be, it's a binding parameter so that Swift will always know if and when this parameter changes.

## Example

In the example below, a `VStack` holds a series of `TextFields` and one `SecureField`. This example resembles a 'Sign Up' page.

```swift
@State private var name = ""
@State private var lastName = ""
@State private var username = ""
@State private var email = ""
@State private var password = ""

var body: some View {
    VStack {
        TextField("Name", text: $name)
        
        TextField("Last Name", text: $lastName)
            .keyboardType(.default)
            
        TextField("Username", text: $username)
            .keyboardType(.asciiCapable)
            
        TextField("Email", text: $email)
            .keyboardType(.emailAddress)
            
        SecureField("Password", text: $password)
    }
    .padding()
}
```

- The top 5 variables are marked as private because it is important that the information does not leave the current page.

> - `TextField` not only has different keyboard types but also keyboard styles. If no type or style is specified, Swift will display them as `(.default)`.

Reviewer 2: What does the default type/style entail?

> - `SecureField` is a `TextField` that instantly hides the character and is commonly used to setup passwords.

Reviewer 2: ```suggestion
- `SecureField` is a `TextField` that instantly hides the character and is commonly used to set up passwords.
```

> There are two images for this example. The first represents empty textfields and the second represents filled textfields:

Reviewer 2: ```suggestion
There are two images for this example. The first represents empty text fields and the second represents filled text fields:
```
